File: rmt/src/test/scala/de/flower/common/validation/UniqueConstraintDetectorTest.scala
package de.flower.common.validation

import org.testng.annotations.Test
import org.testng.Assert._
import javax.persistence.{UniqueConstraint, Table, Column}
import unique.UniqueConstraintDetector

/**
 * 
 * @author oblume
 */

class UniqueConstraintDetectorTest {

    /**
     * Tests that validator analyzes and validates @Column(unique = true) constraints.
     */
    @Test
    def testConstraintDetector() {
        val constraints = UniqueConstraintDetector.detectConstraints(classOf[TestEntity])
        assertEquals(constraints.size(), 2)
    }
}

@Table(uniqueConstraints = Array(new UniqueConstraint(columnNames = Array("col1", "col2"))))
class TestEntity {

    @Column(unique = true)
    var uniqueColumn: String = _

    @Column(unique = false)
    var nonUniqueColumn: String = _

    @Column
    var nonUniqueColumn2: String = _
}

File: rmt/src/test/java/de/flower/common/validation/unique/impl/UniqueConstraintDetectorTest.java
package de.flower.common.validation.unique.impl;

import org.testng.Assert;
import org.testng.annotations.Test;

import javax.persistence.Table;
import javax.persistence.UniqueConstraint;

public class UniqueConstraintDetectorTest {

    @Test
    public void testGetColumnNames() {
        final String[] expectedColumnNames = new String[] { "name", "account_id" };
        final String[] columnNames = UniqueConstraintDetector.getColumnNames("c1", TestEntity.class);
        Assert.assertEquals(expectedColumnNames, columnNames);
    }

    @Table(uniqueConstraints = {
            @UniqueConstraint(name = "c1", columnNames = { "name", "account_id" }),
            @UniqueConstraint(name = "c2", columnNames = { "suId" })
            }
    )
    private static class TestEntity {
    }

}
